src/interrupt.rs

```diff
@@ -34,43 +34,61 @@ pub use bare_metal::{CriticalSection, Mutex, Nr};
 use core::arch::asm;
 
 #[inline]
-/// Disables all interrupts
+/// Disable the global interrupt flag.
 ///
-/// Returns a bool, reflecting whether interrupts were enabled prior to calling this method.
-pub fn disable() -> bool {
```

Comment:
I think a function `is_enabled()` could sometimes be usable. But that function would look at the current `SREG` rather than what is returned from `disable_save()`. I don't really see a use case for getting the interrupt status from *before* `disable_save()`. But if there's a use case, a function interpreting the opaque return type can easily be added.

Reply:
> But that function would look at the current SREG rather than what is returned from disable_save().

Yes, this is what I had in mind as well.

> But if there's a use case, a function interpreting the opaque return type can easily be added.

Right, we could add a method on the opaque type.  But I don't see much value in that right now so I wouldn't bother.  Although maybe it would be nice to have a `Debug`/`uDebug` impl for the type which prints the interrupt status.  This might be useful while debugging...

Reply:
Ok, I implemented both. A new method on the new IrqFlag type to check the flag status and a function is_enabled to check the global flag status, which internally uses the method. That felt quite elegant. But I added a warning comment to the method. I still declared it pub, though.

The Debug/uDebug I didn't do, yet. I'll take a look at it.